### PR TITLE
Add & Edit TE Wrappers

### PR DESCRIPTION
**Pull request by emcelroy**
PT Story: https://www.pivotaltracker.com/story/show/182276802

[#182276802]

### CODE REVIEW

File: lara-typescript/src/section-authoring/components/section-item-plugin-list.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+export const SectionItemPluginList: React.FC<ISectionItemPluginListProps> = ({
+  sectionColumn,
+  sectionId,
+  sectionItemId
+  }: ISectionItemPluginListProps) => {
+  const availablePlugins = useGetAvailablePlugins();
+  const pageItemPlugins = useGetPageItemPlugins(sectionItemId);
+  const pageItemEmbeddableMetaData = useGetPageItemEmbeddableMetaData(sectionItemId);
+  const { userInterface: {editingItemId}, actions: {setEditingItemId}} = React.useContext(UserInterfaceContext);
```
Comment (knowuh) [inline lara-typescript/src/section-authoring/components/section-item-plugin-list.tsx:24]: If you wanted to you could extract `editingItemId` ,`wrappedItemId`, `setEditingItemId`, and `setWrappedItemId` into the same destructured assignment here...

File: lara-typescript/src/section-authoring/components/section-item-plugin-list.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      wrapped_embeddable_type: pageItemEmbeddableMetaData.data?.embeddableType
+    };
+    api.addPageItem(newItem);
+  };
+
+  const renderPluginsMenu = () => {
+    const plugins = availablePlugins.isSuccess ? availablePlugins.data.plugins.map((p: any, i: number) => {
+      const alreadyAdded = pageItemPlugins?.find(pi => pi.name === p.name);
+      if (alreadyAdded) {
+        return null;
```
Comment (knowuh) [inline lara-typescript/src/section-authoring/components/section-item-plugin-list.tsx:55]: This is going to return an array with null objects and components, which is fine, but you could also `.filter` your list, and then `.map` your plugin components, roughly like this:
```
vailablePlugins.data.plugins.filter(p => pageItemPlugins?.find(pi => pi.name === p.name))
.map((p: any, i: number) => ... etc

```

**Review by knowuh — APPROVED** [review]
Looks good to me.

I had some inconsequential suggestions that can be skipped.
👍

**Comment by emcelroy** [comment]
@knowuh Thanks for the suggestions. I made the change regarding the destructuring assignments. I couldn't get the other one about using `.filter` to work. I get the general idea, but couldn't get it to fully work for some unknown reason. Since I'm a bit behind on this work, I'm going to leave things as is in that regard.

**Pull request merged**